releasenotes/notes/fix-as-trafo-grouped-property-iteration-bca2096817b1e497.yaml

```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    Some minor issues in the :class:`~qiskit_nature.transformers.second_quantization.electronic.ActiveSpaceTransformer`
+    were fixed. Namely:
+
+    - iterated properties are no longer deep-copied resulting in non-transformed ones actually being dropped rather than carried on
```

Comment:
Can a simple unit test be done to assert the behavior now is/remains as its now expected.

Reply:
On second thought after my initial "thumb up": adding such a test now would be slightly artificial because the `ASTransformer` actually supports mapping all currently existing properties. This fix is a "backport" from the #640 PR because it also removes some other edge cases in the iteration behavior. I will check again tomorrow with Julian if we can create a unittest for one of those edge cases without needing the Seniority Zero transformer :+1:

Reply:
I added a simple test. It asserts that the `ElectronicBasisTransform` gets updated correctly which it did not prior to this PR (i.e. said test will fail on current `main`). This is does not cover the entirety of this PRs behavioral change but should be sufficient until we have a transformer which can drop certain properties during the transformation.